roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -61,6 +62,61 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
     }
 
     fn handle_declare_mining_job(&mut self, message: DeclareMiningJob) -> Result<SendTo, Error> {
+        {
```

Comment:
put this into a function that takes input a the OldDeclareMiningJob and the Arc<Mutex<JdsMempool>> and removes the old transactions from there

Reply:
Alright